### PR TITLE
feat: Share attachment to kdrive

### PR DESCRIPTION
**Pull request by Ambrdctr**

### CODE REVIEW

File: Mail/Views/Attachment/AttachmentPreview.swift [hunk 1]
```diff
@@ -76,10 +78,58 @@ struct AttachmentPreview: View {
...
+            }
+            try FileManager.default.createDirectory(at: groupContainer, withIntermediateDirectories: false)
+            try FileManager.default.copyItem(
+                at: file,
+                to: destination
+            )
+            return destination
+        } catch {
+            print("Error copying file: \(error.localizedDescription)")
+            return nil
```
Comment (adrien-coye) [inline Mail/Views/Attachment/AttachmentPreview.swift:130]: Do we really want this method to return nil on error ?
If we move this to `ios-core` I think throwing the error can be helpful for the app that would be using this code.
Also we could add a Sentry and remove the print statement.

File: MailShareExtension/Proxy/URLOpener.swift [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+
+public struct URLOpener: URLOpenable {
+    public func canOpen(url: URL) -> Bool {
+        return false
```
Comment (adrien-coye) [inline MailShareExtension/Proxy/URLOpener.swift:23]: Maybe we could print something here in the console to notify the user that this implementation does nothing ?

**Review by adrien-coye — COMMENTED** [review]
PR looks good 👍 

I added a couple of comments regarding code that could be moved to `ios-core`

File: Mail/Proxy/Protocols/URLOpenable.swift [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+
+public protocol URLOpenable {
+    func canOpen(url: URL) -> Bool
```
[the remark this answers is not in the document]
Comment (adrien-coye) [reply Mail/Proxy/Protocols/URLOpenable.swift:22]: Discussed, this would be unnecessary.

**Comment by PhilippeWeidmann** [comment]
Do not merge before kDrive's part is in available in prod.

**Comment by adrien-coye** [comment]
I did a review, it looks good to me. Philippe will give the final go.

**Review by PhilippeWeidmann — DISMISSED** [review]
LGTM.
Once the merge conflict is resolved wait for kDrive before merging.

**Comment by adrien-coye** [comment]
Depends on https://github.com/Infomaniak/ios-kDrive/pull/1087 merged today.
Available in kDrive 4.4.0

**Review by PhilippeWeidmann — APPROVED** [review]

**Review by adrien-coye — APPROVED** [review]
All good 👍

**Pull request merged**